code/modules/mob/animations.dm

```diff
@@ -166,6 +171,8 @@ note dizziness decrements automatically in the mob's Life() proc.
 		if(SOUTHWEST)
 			pixel_x_diff = -8
 			pixel_y_diff = -8
+		else
```

Comment:
i believe it doesn't. Because the animate calls are done with the pixel_x/y_diff vars, which are zeroed if a valid direction isnt found. the return just prevents calling them anyway at a time when they wouldn't do anything, except maybe waste a tiny byte of bandwidth

Attack animations work just fine, the image based thing for mobs isn't affected by the parent return value